addons/dropbox/models.py

```diff
@@ -190,43 +192,38 @@ def get_folders(self, **kwargs):
                 }
             }]
 
-        client = DropboxClient(self.external_account.oauth_key)
-        file_not_found = HTTPError(http.NOT_FOUND, data={
-            'message_short': 'File not found',
-            'message_long': 'The Dropbox file you requested could not be found.'
-        })
-
-        max_retry_error = HTTPError(http.REQUEST_TIMEOUT, data={
-            'message_short': 'Request Timeout',
-            'message_long': 'Dropbox could not be reached at this time.'
-        })
+        client = Dropbox(self.external_account.oauth_key)
 
         try:
-            metadata = client.metadata(folder_id)
-        except ErrorResponse:
-            raise file_not_found
-        except MaxRetryError:
-            raise max_retry_error
-
-        # Raise error if folder was deleted
-        if metadata.get('is_deleted'):
-            raise file_not_found
+            folder_id = '' if folder_id == '/' else folder_id
+            list_folder = client.files_list_folder(folder_id)
+            contents = [x for x in list_folder.entries]
+            while list_folder.has_more:
```

Comment:
@mfraezz , feel free to [check](https://www.dropbox.com/developers/documentation/http/documentation#files-list_folder), but I don't think there is. There's a [`get_metadata`](https://www.dropbox.com/developers/documentation/http/documentation#files-get_metadata) endpoint (which one would think would be the replacement for the old metadata endpoint...) but it only returns metadata for a specific file/folder, not its contents. 

I also can't find anywhere in the documentation how many entries will be returned before `has_more` is set to true and the /list_folder/continue endpoint needs to be used, so I can't say how many requests could potentially be made here. I'm *hoping* it's a very small number since the v1 metadata endpoint returned 10,000 files (see [`file_limit`](https://www.dropbox.com/developers-v1/core/docs#metadata)).